api/cluster/resource/templater_test.go

```diff
@@ -722,6 +751,66 @@ func TestCreateInferenceServiceSpec(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "pyfunc spec with cpu limits specified",
```

Comment:
not expecting changes here but was thinking how can we make this more concise

1. Create default case, the test input should overwrite/merge into the default. Likewise for the output. 
2. Make assert func part of the test input, check for area of interest only (in this cause the assert func checks for cpu limits)

Reply:
> Make assert func part of the test input, check for area of interest only (in this cause the assert func checks for cpu limits)

Yeah this sounds like a great idea, I was thinking of this too, but I'm not too sure if that'll mean we'll leave the door open to other unexpected bugs, e.g. changing the cpu limit stuff causes a totally unlinked change somewhere else in the min/max replicas. But yeah point 1 is probably something necessary that would cut down most of the lines in this file.